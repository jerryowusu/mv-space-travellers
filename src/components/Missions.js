import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { joinMission, leaveMission } from '../redux/missions/missions';
import Mission from './Mission';
import './missions.css';

const Missions = (props) => {
  const { missions } = props;
  const dispatch = useDispatch();

  const joinMissionAction = (id) => {
    dispatch(joinMission(id));
  };

  const cancelMissionAction = (id) => {
    dispatch(leaveMission(id));
  };

  return (
    <table className="missions">
      <thead>
        <tr>
          <th className="bold-text">Mission</th>
          <th className="bold-text">Description</th>
          <th className="bold-text">Status</th>
          <th> </th>
        </tr>
      </thead>
      {
      missions.map((mission) => (
        <tbody key={mission.id} className="mission-item">
          <Mission
            name={mission.name}
            id={mission.id}
            description={mission.description}
            wikipedia={mission.wikipedia}
            joinMission={joinMissionAction}
            leaveMission={cancelMissionAction}
            joined={mission.joined}
          />
        </tbody>
      ))
    }
    </table>
  );
};

Missions.propTypes = {
  missions: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    joined: PropTypes.bool.isRequired,
  })).isRequired,
};

export default Missions;
