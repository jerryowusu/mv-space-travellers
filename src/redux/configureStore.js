import { combineReducers, createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rocketsReducer from './rockets/rockets';

const rootReducer = combineReducers({
  rockets: rocketsReducer,
});

const initialState = {};
const middleware = [thunk];
const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware)),
);

export default store;
