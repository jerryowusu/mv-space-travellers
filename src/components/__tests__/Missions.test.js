import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import { combineReducers, createStore, applyMiddleware } from 'redux';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import rocketsReducer from '../../redux/rockets/rockets';
import missionsReducer from '../../redux/missions/missions';
import Missions from '../Missions';

const mockMissionsData = [
  {
    id: 'mission-1',
    name: 'mission 1',
    description: 'mission-description-1',
    joined: false,
    wikipedia: 'mission-link-1',
  },
  {
    id: 'mission-2',
    name: 'mission 2',
    description: 'mission-description-2',
    joined: false,
    wikipedia: 'mission-link-2',
  },
  {
    id: 'mission-3',
    name: 'mission 3',
    description: 'mission-description-3',
    joined: false,
    wikipedia: 'mission-link-3',
  },
];

const rootReducer = combineReducers({
  rockets: rocketsReducer,
  missions: missionsReducer,
});

const store = createStore(
  rootReducer,
  applyMiddleware(thunk),
);

describe('Missions component tests:', () => {
  it('Missions component renders correctly', () => {
    const missionsPage = renderer.create(
      <Provider store={store}>
        <Missions rockets={mockMissionsData} />
      </Provider>,
    ).toJSON();
    expect(missionsPage).toMatchSnapshot();
  });

  it('Number of rocket elements displayed should be 3', () => {
    const { container } = render(
      <Provider store={store}>
        <Missions rockets={mockMissionsData} />
      </Provider>,
    );
    const page = container.querySelector('.missions');
    expect(page.childElementCount).toBe(3);
  });
});