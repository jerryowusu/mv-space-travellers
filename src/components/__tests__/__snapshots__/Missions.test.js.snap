// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component tests: Missions component renders correctly 1`] = `
<table
  className="missions"
>
  <thead>
    <tr>
      <th
        className="bold-text"
      >
        Mission
      </th>
      <th
        className="bold-text"
      >
        Description
      </th>
      <th
        className="bold-text"
      >
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </thead>
  <tbody
    className="mission-item"
  >
    <tr>
      <td
        className="bold-text"
      >
        mission 1
      </td>
      <td>
        mission-description-1
      </td>
      <td>
        <button
          className="member-btn non-member"
          type="submit"
        >
          Not a Member
        </button>
      </td>
      <td>
        <button
          className="mission-btn join-mission"
          onClick={[Function]}
          type="submit"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
  <tbody
    className="mission-item"
  >
    <tr>
      <td
        className="bold-text"
      >
        mission 2
      </td>
      <td>
        mission-description-2
      </td>
      <td>
        <button
          className="member-btn non-member"
          type="submit"
        >
          Not a Member
        </button>
      </td>
      <td>
        <button
          className="mission-btn join-mission"
          onClick={[Function]}
          type="submit"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
  <tbody
    className="mission-item"
  >
    <tr>
      <td
        className="bold-text"
      >
        mission 3
      </td>
      <td>
        mission-description-3
      </td>
      <td>
        <button
          className="member-btn non-member"
          type="submit"
        >
          Not a Member
        </button>
      </td>
      <td>
        <button
          className="mission-btn join-mission"
          onClick={[Function]}
          type="submit"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
